daemon/cmd/kube_proxy_replacement.go

```diff
@@ -626,7 +626,20 @@ func detectDevices(detectNodePortDevs, detectDirectRoutingDev bool) error {
 			"Cannot retrieve host IP addrs for BPF NodePort device detection")
 	} else {
 		for _, a := range addrs {
-			ifidxByAddr[a.IP.String()] = a.LinkIndex
+			// Any cilium_* interface will never be a valid NodePort or Direct Routing
+			// interface. Skip interface if we cannot resolve it from Netlink via its
+			// ifIndex or if its name begins with cilium_.
+			if link, err := netlink.LinkByIndex(a.LinkIndex); err != nil {
+				log.WithError(err).WithField(logfields.LinkIndex, a.LinkIndex).Warn(
+					"Unable to resolve link from ifIndex, skipping interface")
+				continue
```

Comment:
nit: no need for these `continue`s

Reply:
Thanks for catching that, I've removed them. I had gone back and forth between having an "implicit" and "explicit" `else` condition and the `continue`s would have been required for the implicit condition, but I forgot to remove them after settling on an explicit `else`.